app/controllers/female_missing_infos_controller.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+class FemaleMissingInfosController < PrisonsApplicationController
+  include Wicked::Wizard
+
+  before_action :load_missing_info, except: :new
+
+  # We have to ignore these 2 keys when loading attributes
+  IGNORED_ERROR_KEYS = ['errors', 'validation_context'].freeze
+
+  steps :complexity_level, :delius_information
+
+  def new
+    session[:complexity] = ComplexityForm.new nomis_offender_id: params.fetch(:prisoner_id)
+    session[:case_info] = CaseInformation.new nomis_offender_id: params.fetch(:prisoner_id),
+                                              manual_entry: true
+    if ComplexityMicroService.get_complexity(params.fetch(:prisoner_id))
+      redirect_to wizard_path :delius_information
+    else
+      redirect_to wizard_path(steps.first)
+    end
+  end
+
+  def show
+    @secondary_button =
+      (step == :delius_information) ||
+      (step == :complexity_level && CaseInformation.find_by(nomis_offender_id: @missing_info.nomis_offender_id))
+    render_wizard
+  end
+
+  def update
+    @missing_info.assign_attributes(step_params)
+    save_session
+
+    if @missing_info.valid?
+      if step == :complexity_level
+        ComplexityMicroService.save @missing_info.nomis_offender_id, @missing_info.complexity_level
+        if CaseInformation.find_by nomis_offender_id: @missing_info.nomis_offender_id
```

Comment:
It might be nice to pull this out into its own method:

```ruby
CaseInformation.find_by nomis_offender_id: @missing_info.nomis_offender_id
```

I can see it's been used (at least) a couple of times. Plus I've seen an equivalent call to find out if there's a Complexity level present.

How about refactoring these into private methods?

- `has_complexity_level?`
- `has_case_information?`